### PR TITLE
Add missing entries to the 'build everything' test

### PR DESCRIPTION
**Pull request by cajun-rat**
Change-Id: Iddf0a1939f887aba1a8df345acd907f2a156bd48

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -178,7 +178,7 @@ add_test(NAME test_build_all_off
 
 add_test(NAME test_build_all_on
         COMMAND ${PROJECT_SOURCE_DIR}/tests/build_with_options.sh
-        ${PROJECT_SOURCE_DIR} test_build_all_on "-DBUILD_WITH_CODE_COVERAGE=ON -DBUILD_OSTREE=ON -DBUILD_DEB=ON")
+        ${PROJECT_SOURCE_DIR} test_build_all_on "-DBUILD_WITH_CODE_COVERAGE=ON -DBUILD_OSTREE=ON -DBUILD_DEB=ON -DBUILD_SOTA_TOOLS=ON -DBUILD_ISOTP=ON -DBUILD_OPCUA=ON -DBUILD_LOAD_TESTS=ON")
```
Comment (patrickvacek) [inline tests/CMakeLists.txt]: `BUILD_SYSTEMD`?

Comment (cajun-rat) [reply tests/CMakeLists.txt]: `BUILD_SYSTEMD` is on the default. I could add it though?

Comment (patrickvacek) [reply tests/CMakeLists.txt]: Oh that's right. Porbably don't need to worry about it then.

**Comment by lbonn** [comment]
Since these tests take a long time, I'd be in favor of running less of these tests and instead compiling directly with all options on default CI environment. But in the meantime, it looks good.

Ok when CI is

File: Dockerfile [hunk 2]
```diff
@@ -23,6 +23,7 @@ RUN apt-get update && apt-get -y install \
   lcov \
   libarchive-dev \
   libboost-dev \
+  libboost-iostreams-dev \
```
Comment (patrickvacek) [inline Dockerfile]: Why was this necessary if we ultimately didn't build with BUILD_OPCUA?
